x-pack/legacy/plugins/infra/public/pages/metrics/components/section.tsx

```diff
@@ -31,35 +31,42 @@ export const Section: FunctionComponent<SectionProps> = ({
   stopLiveStreaming,
 }) => {
   const { addNavItem } = useContext(SideNavContext);
-  const subNavItems: SubNavItem[] = [];
 
-  const childrenWithProps = useMemo(
-    () =>
-      Children.map(children, child => {
-        if (isValidElement(child)) {
-          const metric = (metrics && metrics.find(m => m.id === child.props.id)) || null;
-          if (metric) {
-            subNavItems.push({
-              id: child.props.id,
-              name: child.props.label,
-              onClick: () => {
-                const el = document.getElementById(child.props.id);
-                if (el) {
-                  el.scrollIntoView();
-                }
-              },
-            });
-          }
-          return cloneElement(child, {
-            metrics,
-            onChangeRangeTime,
-            isLiveStreaming,
-            stopLiveStreaming,
-          });
-        }
-        return null;
-      }),
-    [children, metrics, onChangeRangeTime, isLiveStreaming, stopLiveStreaming]
```

Comment:
:information_source: Memoizing on `children` had no effect in this case because the parent always supplies a new list of React elements.